api/src/auth/guards/jwt.strategy.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+
+import { ExtractJwt, Strategy } from 'passport-jwt';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor() {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      ignoreExpiration: false,
+      secretOrKey: 'secret',
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using env var for JWT secret</b></div>
<div id="fix">

Consider moving the JWT secret to an environment variable for better security. Using a hardcoded secret in the code might pose a security risk. A similar issue was also found in api/src/auth/auth.module.ts (line 17).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -9,7 +9,7 @@
      super({
        jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
        ignoreExpiration: false,
 -      secretOrKey: 'secret',
 +      secretOrKey: process.env.JWT_SECRET,
      });
    }
 
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/amazon-clone/pull/2#issuecomment-2466146244>#18f66c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged